internal/controller/kbsconfig_controller.go

```diff
@@ -294,6 +295,10 @@ func (r *KbsConfigReconciler) deployOrUpdateKbsDeployment(ctx context.Context) e
 	}
 
 	// Add the kbsFinalizer to the KbsConfig if it doesn't already exist
+	return r.addKbsConfigFinalizer(ctx)
```

Comment:
The finalizer needs to be added only during the creation. So this can be removed.